aicsimageio/aics_image.py

```diff
@@ -0,0 +1,418 @@
+            attempt to parse dim order.
+        kwargs: Any
+            Extra keyword arguments that will be passed down to the reader subclass.
+
+        Examples
+        --------
+        Initialize an image and read the slices specified as a numpy array.
+
+        >>> img = AICSImage("my_file.tiff")
+        ... zstack_t8 = img.get_image_data("ZYX", S=0, T=8, C=0)
+
+        Initialize an image, construct a delayed dask array for certain slices, then
+        read the data.
+
+        >>> img = AICSImage("my_file.czi")
+        ... zstack_t8 = img.get_image_dask_data("ZYX", S=0, T=8, C=0)
+        ... zstack_t8_data = zstack_t8.compute()
+
+        Initialize an image with a dask or numpy array.
+
+        >>> data = np.random.rand(100, 100)
+        ... img = AICSImage(data)
+
+        Initialize an image from S3 with s3fs.
+
+        >>> fs = s3fs.S3FileSystem("s3://my_bucket")
+        ... f = fs.open("s3://my_bucket/my_file.tiff")
+        ... img = AICSImage(f)
+
+        Initialize an image and pass arguments to the reader using kwargs.
+
+        >>> img = AICSImage("my_file.czi", chunk_by_dims=["T", "Y", "X"])
+
+        Notes
+        -----
+        Constructor for AICSImage class intended for providing a unified interface for
+        dealing with microscopy images. To extend support to a new reader simply add a
+        new reader child class of Reader ([readers/reader.py]) and add the class to
+        SUPPORTED_READERS variable.
+        """
+        pass
+
+    @staticmethod
+    def determine_reader(image: types.ImageLike) -> Reader:
+        """
+        Cheaply check to see if a given file is a recognized type and return the
+        appropriate reader for the file.
+        """
+        pass
+
+    @property
+    def reader(self) -> Reader:
+        """
+        This property returns the object created to read the image file type.
+        The intent is that if the AICSImage class doesn't provide a raw enough
+        interface then the base class can be used directly.
+        """
+        pass
+
+    @property
+    def dask_data(self) -> da.Array:
+        """
+        Returns the image as a dask array with dimension ordering "STCZYX".
+        """
+        pass
+
+    @property
+    def data(self) -> np.ndarray:
+        """
+        Return the image as a numpy array with dimension ordering "STCZYX".
+        """
+        pass
+
+    @property
+    def dtype(self) -> np.dtype:
+        """
+        Data-type of the image array's elements.
+        """
+        pass
+
+    @property
+    def shape(self) -> Tuple[int]:
+        """
+        Tuple of the image array's dimensions.
+        """
+        pass
+
+    @property
+    def dims(self) -> Dimensions:
+        """
+        Object with the paired dimension names and their sizes.
+        """
+        pass
+
+    def get_image_dask_data(
+        self, out_orientation: Optional[str] = None, **kwargs
```

Comment:
suggestion: out_orientation feels just a bit vague to me... could be requested_dimension_order ?

Reply:
I like and am okay with `out_dimension_order`, `dimension_order_out`, and simply `dimension_order`. Thoughts?

Reply:
Voted 2/3 `dimension_order_out`, Dan abstains